tosfs/core.py

```diff
@@ -203,6 +203,87 @@ def info(
 
         return self._try_dir_info(bucket, key, path, fullpath)
 
+    def rmdir(self, path: str) -> None:
+        """Remove a directory if it is empty.
+
+        Parameters
+        ----------
+        path : str
+            The path of the directory to remove. The path should be in the format
+            `tos://bucket_name/directory_path`.
+
+        Raises
+        ------
+        FileNotFoundError
+            If the directory does not exist.
+        NotADirectoryError
+            If the path is not a directory.
+        TosfsError
+            If the directory is not empty,
+             or if there is an error during the removal process.
+
+        Examples
+        --------
+        >>> fs = TosFileSystem()
+        >>> fs.rmdir("tos://mybucket/mydir/")
+
+        """
+        path = self._strip_protocol(path).rstrip("/") + "/"
+        bucket, key, _ = self._split_path(path)
+        if not key:
+            return self._rm_bucket(path)
```

Comment:
This is a dangerous action. s3fs also supports it. Also, I carried out some defensive verification checks as soon as possible, but risks still exist.
Consider adding an env flag to let users make decisions. WDYT? @openinx @xianyinxin

Reply:
Compared aliging to s3, i think it would be better to block users from doing this. We could refere to s3 but not should to be compatible with s3.

Reply:
Sounds reasonable, so if can not get a key, throw an exception?